frontend/src/views/Abfrage.vue

```diff
@@ -331,41 +345,35 @@ export default class Abfrage extends Mixins(
     this.openAbfragevariantenFormular = false;
   }
 
-  private saveAbfrageInStore(abfrage: InfrastrukturabfrageModel) {
-    this.$store.commit("search/selectedAbfrage", _.cloneDeep(abfrage));
-  }
-
-  private async abfrageFreigeben(): Promise<void> {
-    if (this.validate()) {
-      if (
-        this.abfrageWrapped.infrastrukturabfrage.abfrage.statusAbfrage ===
-        AbfrageListElementDtoStatusAbfrageEnum.Angelegt
-      ) {
-        this.infrastrukturabfrageFreigeben();
-      } else {
-        this.showWarningInInformationList('Die Abfrage muss den Status "angelegt" besitzen');
-      }
-    } else {
-      this.showWarningInInformationList("Es gibt noch Validierungsfehler");
-    }
-  }
-
-  private async infrastrukturabfrageFreigeben(): Promise<void> {
+  private async startStatusUebergang(transition: TransitionDto) {
     const validationMessage: string | null = this.findFaultInInfrastrukturabfrageForSave(
       this.abfrageWrapped.infrastrukturabfrage
     );
     if (_.isNil(validationMessage)) {
       await this.updateInfrastrukturabfrage(this.abfrageWrapped.infrastrukturabfrage, true);
-      this.freigabInfrastrukturabfrage(this.abfrageWrapped.infrastrukturabfrage.id as string, true).then(() => {
-        this.returnToUebersicht("Die Abfrage wurde erfolgreich freigegeben", Levels.SUCCESS);
-      });
-      this.openAbfrageFormular = true;
-      this.openAbfragevariantenFormular = false;
+      const requestSuccessful = await this.statusUebergangRequest(transition, this.abfrageId);
+      if (requestSuccessful) {
+        if (!(transition.buttonName === "IN BEARBEITUNG SETZEN")) {
```

Comment:
Es ist unpraktisch, nach einem statischen String abzufragen (z.B. wenn sich der Wert in der Zukunft ändert).

Du kannst im Backend bei den `StatusAbfrageEvents` die Annotation `@Schema(enumAsRef = true)` setzen, damit das Enum als API-Klasse generiert wird, siehe auch [hier](https://github.com/it-at-m/isi-backend/blob/ISI-480-Statuswechsel-in-einer-Infrastrukturabfrage/src/main/java/de/muenchen/isi/infrastructure/entity/enums/lookup/UncertainBoolean.java#L11).